acceptance/monotonic_insert_test.go

```diff
@@ -0,0 +1,208 @@
+	}
+	if _, err := clients[0].Exec(`
+CREATE DATABASE mono;
+CREATE TABLE IF NOT EXISTS mono.mono (val INT, sts STRING, node INT, tb INT);
+INSERT INTO mono.mono VALUES(-1, '0', -1, -1)`); err != nil {
+		t.Fatal(err)
+	}
+
+	var idGen uint64
+
+	invoke := func(client mtClient) {
+		logPrefix := fmt.Sprintf("%03d.%03d: ", atomic.AddUint64(&idGen, 1), client.ID)
+		l := func(msg string, args ...interface{}) {
+			log.Infof(logPrefix+msg, args...)
+		}
+		l("begin")
+		defer l("done")
+
+		var exRow, insRow mtRow
+		var attempt int
+		if err := crdb.ExecuteTx(client.DB, func(tx *sql.Tx) error {
+			attempt++
+			l("attempt %d", attempt)
+			if err := tx.QueryRow(`SELECT cluster_logical_timestamp()`).Scan(
+				&insRow.sts,
+			); err != nil {
+				l(err.Error())
+				return err
+			}
+
+			l("read max val")
+			if err := tx.QueryRow(`SELECT MAX(val) AS m FROM mono.mono`).Scan(
+				&exRow.val,
+			); err != nil {
+				l(err.Error())
+				return err
+			}
+
+			l("read max row for val=%d", exRow.val)
+			if err := tx.QueryRow(`SELECT sts, node, tb FROM mono.mono WHERE val = $1`,
+				exRow.val,
+			).Scan(
+				&exRow.sts, &exRow.node, &exRow.tb,
+			); err != nil {
+				l(err.Error())
+				return err
+			}
+
+			l("insert")
+			if err := tx.QueryRow(`
+INSERT INTO mono.mono (val, sts, node, tb) VALUES($1, $2, $3, $4)
+RETURNING val, sts, node, tb`,
+				exRow.val+1, insRow.sts, client.ID, 0,
+			).Scan(
+				&insRow.val, &insRow.sts, &insRow.node, &insRow.tb,
+			); err != nil {
+				l(err.Error())
+				return err
+			}
+			l("commit")
+			return nil
+		}); err != nil {
+			t.Errorf("%T: %v", err, err)
+		}
+	}
+
+	verify := func() {
+		client := clients[0]
+		var numDistinct int
+		if err := client.QueryRow("SELECT COUNT(DISTINCT(val)) FROM mono.mono").Scan(
+			&numDistinct,
+		); err != nil {
+			t.Fatal(err)
+		}
+		rows, err := client.Query("SELECT val, sts, node, tb FROM mono.mono ORDER BY val ASC, sts ASC")
+		if err != nil {
+			t.Fatal(err)
+		}
+		var results mtRows
+		for rows.Next() {
+			var row mtRow
+			if err := rows.Scan(&row.val, &row.sts, &row.node, &row.tb); err != nil {
+				t.Fatal(err)
+			}
+			results = append(results, row)
+		}
+
+		if !sort.IsSorted(results) {
+			t.Error("results are not sorted")
+		}
+
+		if numDistinct != len(results) {
+			t.Error("'val' column is not unique: %d results, but %d distinct",
+				len(results), numDistinct)
+		}
+
+		fmt.Println(results)
+	}
+
+	const concurrency = 10
+
+	sem := make(chan struct{}, concurrency)
+	timer := time.After(cfg.Duration)
+
+	defer verify()
+	defer func() {
+		// Now that consuming has stopped, fill up the semaphore (i.e. wait for
+		// still-running goroutines to stop)
+		for i := 0; i < concurrency; i++ {
+			sem <- struct{}{}
+		}
+	}()
+
+	for {
+		select {
+		case sem <- struct{}{}:
+		case <-stopper:
+			return
+		case <-timer:
+			return
+		}
+		go func(client mtClient) {
+			invoke(client)
+			<-sem
+		}(clients[rand.Intn(c.NumNodes())])
```

Comment:
I'd suggest NumNodes different goroutines, each running txns in a (local) loop to a separate note. That would be more like the original workload.